.github/workflows/publish-engine.yml

```diff
@@ -0,0 +1,28 @@
+name: publish-engine
```

Comment:
All the config in here is just my best guess at what should work; I'm not actually sure how to test it before we actually do a new engine tag. Let me know if there's a good way of going about that.

Primary risks:
1. The `tags` matching patterns are wrong (just guessing based off reading the gha docs)
2. `secrets.RELEASE_DAGGER_CI_TOKEN` doesn't have the permissions needed to push to ghcr (https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)